typescript-react/src/components/CardList.tsx

```diff
@@ -0,0 +1,129 @@
+import { formatDate, getTimeAgo } from "../utils/time";
+import { useState, MouseEvent, useEffect } from "react";
+import { transCardLinkData } from "../utils/transCardLinkData";
+import KebabModal from "./Modal/KebabModal";
+import "../style/CardList.css";
+
+interface Link {
+  id: number;
+  createdAt: number;
+  url: string;
+  title: string;
+  description: string;
+  imageSource: string | null;
+  folder: string;
+}
+
+interface CardListProps {
+  cardlinks: {
+    links?: Link[];
+  };
+  searchTerm: string;
+  selectedFolderName: string;
+}
+
+function CardList({
+  cardlinks,
+  searchTerm,
+  selectedFolderName,
+}: CardListProps) {
+  const transformedData = transCardLinkData(cardlinks);
+  const allLinks: Link[] = transformedData.links ?? [];
+
+  const [isPopoverOpen, setIsPopoverOpen] = useState(false);
+  const [filteredLinks, setFilteredLinks] = useState<Link[]>([]);
+
+  const handleOpenPopover = (
+    event: MouseEvent<HTMLButtonElement, { stopPropagation: () => void }>
+  ) => {
+    event.stopPropagation();
+    setIsPopoverOpen(true);
+  };
+
+  const handleClosePopover = () => {
+    setIsPopoverOpen(false);
+  };
+
+  const handleCardClick = (link: Link) => {
+    window.open(link.url, "_blank");
+  };
+
+  const handleSearch = (searchTerm: string) => {
+    const filtered = allLinks.filter(
+      link =>
+        (link.folder === selectedFolderName &&
+          link.url.toLowerCase().includes(searchTerm.toLowerCase())) ||
+        (link.title &&
+          link.title.toLowerCase().includes(searchTerm.toLowerCase())) ||
+        (link.description &&
+          link.description.toLowerCase().includes(searchTerm.toLowerCase()))
+    );
+
+    console.log(filtered);
+
+    setFilteredLinks(filtered);
+  };
```

Comment:
조건식의 중첩이 많아지면 보기 어려워지는데, 코드가 구체적으로 적혀있어서 코드 한줄한줄 어떤 의미인지 생각해 가면서 읽어야 하겠네여. 
이럴때는 구체적인 코드를 변수에 담아서 특정 이름으로 쉽게 구분할 수 있게 코드단위로 의미 부여 해주는 리팩토링을 해주면 좋습니다.

```suggestion
const handleSearch = (searchTerm: string) => {
  const searchTermLower = searchTerm.toLowerCase();

  const isMatch = (link: Link) => {
    const titleMatch = link.title?.toLowerCase().includes(searchTermLower);
    const descriptionMatch = link.description?.toLowerCase().includes(searchTermLower);
    const urlMatch = link.url.toLowerCase().includes(searchTermLower);
    const isInSelectedFolder = link.folder === selectedFolderName;

    return (isInSelectedFolder && urlMatch) || titleMatch || descriptionMatch;
  };

  const filteredLinks = allLinks.filter(isMatch);

  console.log(filteredLinks);

  setFilteredLinks(filteredLinks);
};

```

그리고 filter메서드가 받는 콜백 함수를 기명 함수로 만들어서 전달하면 콜백함수가 어떤 역할을 하는지도 의미가 명확해집니다.